entgql/annotation.go

```diff
@@ -335,6 +350,35 @@ func (a queryFieldAnnotation) Directives(directives ...Directive) queryFieldAnno
 	return a
 }
 
+type MutationOption interface {
+	IsCreate() bool
+}
+
+type builtinMutation bool
+
+func (v builtinMutation) IsCreate() bool { return bool(v) }
```

Comment:
Why not use functional options that can change the `Options` object (instead of interface). Feels easier to write.